### DISCUSSION TITLE
Difference in counting the number of rules from GIT

### DISCUSSION THREAD

**Discussion by shweta13agg** [community/community#24429] [Apps, API and Webhooks]
<p>I’m doing Automation to count the total number of rules from the GIT by using the GIT Bash Regular Expression.</p>
<ol>
<li>Firstly I connected to the GIT Project remotely.</li>
<li>Used the below expression ( git log --numstat |egrep “.brl$” | awk ‘{for(i=3; i&lt;=NF; ++i) printf "%s ", $i; print “”}’ | sort -u | wc -l ) to find the count of rules (extension ending with “.dslr”)</li>
<li>But I’m getting the difference in count, e.g., if some rules are in expired folder, it shows those rules in expired folder and other folder as well as shown the highlighted rule as below <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8b5a445bcb8ca1ee86a0549b832db9fe48fe3706.jpeg" data-download-href="/uploads/short-url/jSLLp8iXU2PqjatwHeYsXiiEiXk.jpg?dl=1" title="Capture.JPG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/50583102/181084709-6240f370-93b6-493e-a0b4-962bb004fdca.jpeg" alt="Capture.JPG" data-base62-sha1="jSLLp8iXU2PqjatwHeYsXiiEiXk" width="690" height="172" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8b5a445bcb8ca1ee86a0549b832db9fe48fe3706_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture.JPG</span><span class="informations">821×205 70.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</li>
<li>But when I manually check into the particular folder, 40078 rule is present only in expired folder.</li>
<li>Can anyone help me why this issue is occurring, Is git store some cache or something?</li>
</ol>

**Comment by lee-dohm** [comment]
<p>This sounds more like you’re needing help with debugging your one line bash script than a question about git? Can you give us an example of exactly what is not working that is separate from your bash script? Also, it would be helpful if you had a public repository where you could reproduce the problem that we could take a look at.</p>
<p>Let us know!</p>

**Comment by shweta13agg** [comment]
<p>Hey <a class="mention" href="/lee-dohm">@lee-dohm</a> ,</p>
<p>Actually I’m counting the number of files ending with particular extension (e.g. extension can be .brl or .dslr) in a project. So, currently I’m able to find the count of files from git itself by running that command which I mentioned previuosly.</p>
<p>But the problem is there are some files say F1.dslr and F2.dslr in Folder1 and Folder2 respectively.</p>
<p>Previuosly F1.dslr file was in Folder1 but now it is moved to Folder 2.</p>
<p>So when I’m counting the files by that command, it shows the F1.dslr file is in both Folder 1&amp;2.</p>
<p>But in actual scenario F1.dslr is availbale only in Folder2.</p>
<p>So, my concern is while counting the files, why it shows the F1.dslr file in both folder?</p>

**Comment by lee-dohm** [comment, marked as answer]
<blockquote>
<p>But the problem is there are some files say F1.dslr and F2.dslr in Folder1 and Folder2 respectively.</p>
</blockquote>
<blockquote>
<p>Previuosly F1.dslr file was in Folder1 but now it is moved to Folder 2.</p>
</blockquote>
<blockquote>
<p>So when I’m counting the files by that command, it shows the F1.dslr file is in both Folder 1&amp;2.</p>
</blockquote>
<p>The reason for this is that you started your command with <code>git log --numstat</code>. This produces the list of every change that has ever been made to your git repository, including every file that does or ever has existed in your repo. If you only want the files that currently exist in your repository, then you should use the <code>ls</code> command just like you would with any other directory. Or if you really only want to see the files that git is tracking, you can use the <code>git ls-files</code> command.</p>
<p>I hope that helps!</p>

**Comment by shweta13agg** [comment]
<p>Thank you soo much <a class="mention" href="/lee-dohm">@lee-dohm</a> . It’s working now :slight_smile:</p>
<p>Thanks a lott.</p>
